v2/kafka-to-bigquery/src/main/java/com/google/cloud/teleport/v2/transforms/BigQueryDynamicDestination.java

```diff
@@ -58,8 +59,18 @@ public GenericRecord getDestination(ValueInSingleWindow<KV<GenericRecord, TableR
 
   @Override
   public TableDestination getTable(GenericRecord element) {
-    // tablename + record name (same across schemas) + schema id?
-    String bqQualifiedFullName = element.getSchema().getFullName().replace(".", "-");
+    Pattern sanitizePattern = Pattern.compile("[^\\w- ]", Pattern.UNICODE_CHARACTER_CLASS);
+    String sanitizedNamespace =
+        sanitizePattern.matcher(element.getSchema().getNamespace()).replaceAll("-");
+    String sanitizedName = sanitizePattern.matcher(element.getSchema().getName()).replaceAll("-");
+
+    Pattern multiHyphenRemovalPattern = Pattern.compile("-+", Pattern.UNICODE_CHARACTER_CLASS);
+    sanitizedNamespace = multiHyphenRemovalPattern.matcher(sanitizedNamespace).replaceAll("-");
+    sanitizedName = multiHyphenRemovalPattern.matcher(sanitizedName).replaceAll("-");
```

Comment:
Consider moving the sanitation logic above to a utils class.

Reply:
done